### ISSUE TITLE
Undocumented effect of reseting inner pointer of the array to 0 index in group of std methods

### ISSUE THREAD

**Issue by Salomon88** [php/doc-en#876]
In practice were found a group of functions that after their invocation reset the array pointer to 0 position. Unfortunately, this effect isn't documented and can be unexpected. The example below is a little bit synthetic, but it can be more complex cases:
```
<?php

$arr = array(1, 7, 2);
echo current($arr);
next($arr); //Moves to 1 index
sort($arr); //Rewinds to 0 index
echo current($arr);
```
List of functions to be documented:
rsort, shuffle, uasort, uksort, sort, krsort, ksort, natcasesort, natsort, array_multisort, arsort, asort, array_unshift

**Comment by Salomon88** [comment]
Example - https://3v4l.org/beLfO

**Comment by cmb69** [comment]
Closes via 6a6f43d1c490a57b452656db285de6d136055ed2.
